### PR TITLE
ci: migrate to mage-tools

### PR DESCRIPTION
**Pull request by odsod**

### CODE REVIEW

File: routingv7/calculatematrix.go [hunk 1]
```diff
@@ -58,8 +58,7 @@ func (r *CalculateMatrixRequest) QueryString() string {
 	for i, wp := range r.DestinationWaypoints {
 		values.Add(fmt.Sprintf("destination%d", i), wp.QueryString())
 	}
-	mode := r.Mode.String()
-	if mode != "" {
+	if mode := r.Mode.String(); mode != "" {
```
Comment (pmarkus) [inline routingv7/calculatematrix.go:61]:
```suggestion
	if r.Mode.String() != "" {
```

File: routingv7/client.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ type service struct {
 }
 
 // An ErrorResponse reports the error caused by an API request.
+// nolint: errname // TODO: Consider breaking API change to follow XxxError convention.
```
Comment (pmarkus) [inline routingv7/client.go:46]:
```suggestion
// nolint: errname // TODO(oscar.soderlund@einride.tech): Consider breaking API change to follow XxxError convention.
```

Comment (pmarkus) [reply routingv7/client.go:46]: Suggesting adding email to TODO:s for traceability

Comment (odsod) [reply routingv7/client.go:46]: IMO `git blame` should be enough

Comment (odsod) [reply routingv7/client.go:46]: this is a public repo, so would rather not put email in

File: routingv8/client.go [hunk 3]
```diff
@@ -37,13 +37,13 @@ type service struct {
 	Client *Client
 }
 
-// An errorResponse reports the error caused by an API request.
-type errorResponse struct {
+// An responseError reports the error caused by an API request.
```
Comment (pmarkus) [inline routingv8/client.go:40]:
```suggestion
// A responseError reports the error caused by an API request.
```

File: routingv8/client.go [hunk 4]
```diff
@@ -143,6 +143,6 @@ func checkResponse(r *http.Response) error {
 	if err != nil {
 		return err
 	}
-	errorResponse := &errorResponse{Response: &response}
+	errorResponse := &responseError{Response: &response}
 	return errorResponse
```
Comment (pmarkus) [inline routingv8/client.go:147]:
```suggestion
	err = &responseError{Response: &response}
	return err
```

**Review by pmarkus — COMMENTED** [review]
Left some small suggestions, otherwise LGTM

File: routingv7/client.go (hunk 2 above)
Comment (pmarkus) [reply routingv7/client.go:46]: I'm reverting this suggestion. Had a further look, and Google was now discouraging this style. Possible that they put an issue tracker link in the TODO parenthesis instead.

File: routingv7/calculatematrix.go (hunk 1 above)
Comment (odsod) [reply routingv7/calculatematrix.go:61]: good point :smile_cat: fixed

File: routingv7/calculateroute.go [hunk 5]
```diff
@@ -50,8 +50,7 @@ func (r *CalculateRouteRequest) QueryString() string {
 	for i, wp := range r.Waypoints {
 		values.Add(fmt.Sprintf("waypoint%d", i), wp.QueryString())
 	}
-	mode := r.Mode.String()
-	if mode != "" {
+	if mode := r.Mode.String(); mode != "" {
```
[the remark this answers is not in the document]
Comment (odsod) [reply routingv7/calculateroute.go:53]: fixed!

File: routingv8/client.go (hunk 3 above)
Comment (odsod) [reply routingv8/client.go:40]: fixed!

File: routingv8/client.go (hunk 4 above)
Comment (odsod) [reply routingv8/client.go:147]: ah we can just inline this

**Comment by odsod** [comment]
@pmarkus Great comments - fixed and pushed!

**Review by pmarkus — APPROVED** [review]

**Pull request merged**